SingularityService/src/main/java/com/hubspot/singularity/data/SingularityValidator.java

```diff
@@ -165,6 +166,12 @@ public SingularityRequest checkSingularityRequest(SingularityRequest request, Op
       checkBadRequest(!request.getScheduleType().isPresent(), "ScheduleType can only be set for scheduled requests");
     }
 
+    if (request.getScheduleTimeZone().isPresent()) {
+      if (!ArrayUtils.contains(TimeZone.getAvailableIDs(), request.getScheduleTimeZone().get())) {
+        badRequest("scheduleTimeZone %s was not a valid timezone code (e.g. 'US/Eastern' or 'GMT')", request.getScheduleTimeZone().get());
```

Comment:
As far as I can tell, the getAvailableIds() list is maintained by Oracle, not necessarily conforming to any particular standard (that I can find).

https://garygregory.wordpress.com/2013/06/18/what-are-the-java-timezone-ids/